### PR TITLE
Weapon Detection System

### PR DESCRIPTION
**Pull request by venkat-2811**
### Requesting to submit a pull request to the PyVerse repository.

---

#### Issue Title
**Please enter the title of the issue related to your pull request.**  
*Weapon Detection System*

- [X] I have provided the issue title.

---

#### Info about the Related Issue
**What's the goal of the project?**  
*The goal of the project is to create a real-time weapon detection system using the YOLOv3 algorithm. The system can detect weapons in real-time from a webcam feed or video file and display the results with bounding boxes and labels.*

- [X] I have described the aim of the project.

---

#### Name
**Please mention your name.**  
*Sunkara Venkata Karthik Sai*

- [X] I have provided my name.

---

#### GitHub ID
**Please mention your GitHub ID.**  
*venkat-2811*

- [X] I have provided my GitHub ID.

---

#### Email ID
**Please mention your email ID for further communication.**  
*venkatakarthiksai.s@gmail.com*
- [X] I have provided my email ID.

---

#### Identify Yourself
**Mention in which program you are contributing (e.g., WoB, GSSOC, SSOC, SWOC).**  
*GSSOC*

- [X] I have mentioned my participant role.

---

#### Closes
**Enter the issue number that will be closed through this PR.**  
*Closes: #620 *

- [X] I have provided the issue number.

---

#### Describe the Add-ons or Changes You've Made
**Give a clear description of what you have added or modified.**  
*I have added a real-time weapon detection system using the YOLOv3 algorithm. It supports both webcam and video file detection. The system is customizable with options to adjust detection thresholds and displays bounding boxes and labels around detected weapons.*

- [X] I have described my changes.

---

#### Type of Change
**Select the type of change:**  
- [ ] Bug fix (non-breaking change which fixes an issue)  
- [X] New feature (non-breaking change which adds functionality)  
- [ ] Code style update (formatting, local variables)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] This change requires a documentation update  

---

#### How Has This Been Tested?
**Describe how your changes have been tested.**  
*The system has been tested using both a live webcam feed and a video file. Multiple weapons were successfully detected in real-time, and the bounding boxes and labels were accurately displayed. Custom detection thresholds were also tested for performance optimization.*

- [X] I have described my testing process.

---

#### Checklist
**Please confirm the following:**  
- [X] My code follows the guidelines of this project.  
- [X] I have performed a self-review of my own code.  
- [X] I have commented my code, particularly wherever it was hard to understand.  
- [X] I have made corresponding changes to the documentation.  
- [X] My changes generate no new warnings.  
- [X] I have added things that prove my fix is effective or that my feature works.  
- [X] Any dependent changes have been merged and published in downstream modules.

---

Let me know if you need any further modifications!

### CODE REVIEW

File: Computer Vision/WEAPON-DETECTION-SYSTEM/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+### Setup
+
+1.  **Clone the repository and navigate to the project directory**:
+
+    ```bash
+    git clone <repository_url>
+    cd <repository_name>
+
+2.  **Download YOLOv3 weights and configuration files**:
```
Comment (ruhi47) [inline Computer Vision/WEAPON-DETECTION-SYSTEM/README.md:52]: Hey @venkat-2811 ,
Please include the links for the `yolov3.weights` and 'yolov3.cfg'   download.

File: Computer Vision/WEAPON-DETECTION-SYSTEM/weapon_detection.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import numpy as np
+
+# Load Yolo
+net = cv2.dnn.readNet("yolov3_training_2000.weights", "yolov3_testing.cfg")
+net.setPreferableBackend(cv2.dnn.DNN_BACKEND_DEFAULT)
+net.setPreferableTarget(cv2.dnn.DNN_TARGET_CPU)
+classes = ["Weapon"]
+
+# Open the video file
+cap = cv2.VideoCapture("pistol1.mp4")
```
Comment (ruhi47) [inline Computer Vision/WEAPON-DETECTION-SYSTEM/weapon_detection.py:11]: Also, include some videos that include pistol and some without it to see the accuracy and performance of the model. Have at least 3 videos in the folder.

**Review by ruhi47 — CHANGES_REQUESTED** [review]
Do the above mentioned changes and will review again.
Should you need any help, do not hesitate to ask questions!
Thank you !

File: Computer Vision/WEAPON-DETECTION-SYSTEM/.gitattributes [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Auto detect text files and perform LF normalization
```
[the remark this answers is not in the document]
Comment (venkat-2811) [reply Computer Vision/WEAPON-DETECTION-SYSTEM/.gitattributes:1]: Okay

**Pull request closed without merging**